nibabel/freesurfer/mghformat.py

```diff
@@ -13,30 +13,38 @@
 from os.path import splitext
 import numpy as np
 
+from ..affines import voxel_sizes
 from ..volumeutils import (array_to_file, array_from_file, Recoder)
 from ..spatialimages import HeaderDataError, SpatialImage
-from ..fileholders import FileHolder, copy_file_map
-from ..arrayproxy import ArrayProxy
+from ..fileholders import FileHolder
+from ..arrayproxy import ArrayProxy, reshape_dataobj
 from ..keywordonly import kw_only_meth
 from ..openers import ImageOpener
+from ..wrapstruct import LabeledWrapStruct
 
 # mgh header
 # See https://surfer.nmr.mgh.harvard.edu/fswiki/FsTutorial/MghFormat
 DATA_OFFSET = 284
 # Note that mgh data is strictly big endian ( hence the > sign )
 header_dtd = [
-    ('version', '>i4'),
-    ('dims', '>i4', (4,)),
-    ('type', '>i4'),
-    ('dof', '>i4'),
-    ('goodRASFlag', '>i2'),
-    ('delta', '>f4', (3,)),
-    ('Mdc', '>f4', (3, 3)),
-    ('Pxyz_c', '>f4', (3,))
+    ('version', '>i4'),             # 0; must be 1
+    ('dims', '>i4', (4,)),          # 4; width, height, depth, nframes
+    ('type', '>i4'),                # 20; data type
+    ('dof', '>i4'),                 # 24; degrees of freedom
+    ('ras_good', '>i2'),            # 28; *_ras fields valid
+    ('voxelsize', '>f4', (3,)),     # 30; zooms (X, Y, Z)
+    ('x_ras', '>f4', (3, 1)),       # 42; X direction cosine column
```

Comment:
By including all three columns as a single, row-ordered `Mdc` matrix, we're exposing a transposed matrix. It was being transposed correctly in and out, but if someone does decide to use it directly, they need to know this. I prefer explicit to silently misleading...

I could use `(3,)` if that's important. But it will mean users *knowing* that they're columns and not row vectors (can add to the docs).

For context, this is the comparable portion of the C data structure, which is just 16 individual `float` fields: https://github.com/freesurfer/freesurfer/blob/5367eec84621b331271e97f4e1f1502c3f6d0d87/include/mri.h#L184-L191

Reply:
OK - fine to leave as is (3,1) etc.